check_copyright.py

```diff
@@ -282,8 +282,12 @@ def has_valid_copyright(file_name: str, mime: str, is_on_ignore: bool, config_se
                     template = '/* SPDX-FileCopyrightText: ' + config_section['espressif_copyright']
                 if mime == MIME['python']:
                     template = '# SPDX-FileCopyrightText: ' + config_section['espressif_copyright']
-                candidate_line = template.format(years=format_years(years[0], file_name))
-                no_time_update = template.format(years=format_years(years[0], file_name, years[1] or years[0]))
+                if is_new_file:
```

Comment:
Perhaps it would be easier just to do once 
```python
if is_new_file:
    years[0] = 0
```
and the no need to have here these big if-else blocks.

Reply:
Added the following change instead to the same effect - 
```
                if is_new_file:
                    years = (0, None)
                else:
                    years = extract_years_from_espressif_notice(matches.group(1))
```